charts/shoot-core/charts/csi-packet/templates/controller.yaml

```diff
@@ -0,0 +1,88 @@
+kind: Service
+apiVersion: v1
+metadata:
+  name: csi-packet-pd
+  labels:
+    app: csi-packet-pd
+spec:
+  selector:
+    app: csi-packet-pd
+  ports:
+    - name: dummy
+      port: 12345
+
+---
+
+kind: StatefulSet
+apiVersion: apps/v1beta1
+metadata:
+  name: csi-packet-controller
+  namespace: kube-system
+spec:
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      partition: 0
+  serviceName: "csi-packet-pd"
+  replicas: 1
+  selector:
+    matchLabels:
+      app: csi-packet-pd-driver
+  template:
+    metadata:
+      labels:
+        app: csi-packet-pd-driver
```

Comment:
You are mounting a secret as volume here, so you should add checksum of it here to allow automatic restart of the container if the secret changes, see e.g., https://github.com/gardener/gardener/blob/master/charts/shoot-core/charts/csi-alicloud/templates/csi-diskplugin-ds.yaml#L19

Reply:
I like the idea of leveraging helm's checksumming to do that. Will look at that.

Separately, looking at the Alicloud example you linked to, I see the checksum, but where do you see a secret mounted there as a volume? The entire volume set is:

```yml
      volumes:
        - name: registration-dir
          hostPath:
            path: /var/lib/kubelet/plugins_registry
            type: DirectoryOrCreate
        - name: plugin-dir
          hostPath:
            path: /var/lib/kubelet/plugins/diskplugin.csi.alibabacloud.com
            type: DirectoryOrCreate
        - name: pods-mount-dir
          hostPath:
            path: /var/lib/kubelet
            type: Directory
        - name: host-dev
          hostPath:
            path: /dev
```

Reply:
Added